packages/vite/src/node/plugins/index.ts

```diff
@@ -161,3 +164,14 @@ export function getHookHandler<T extends ObjectHook<Function>>(
 ): HookHandler<T> {
   return (typeof hook === 'object' ? hook.handler : hook) as HookHandler<T>
 }
+
+// Same as `@rollup/plugin-alias` default resolver, but we attach additional meta
+// if we can't resolve to something, which will error in `importAnalysis`
+export const viteAliasCustomResolver: ResolverFunction = async function (
+  id,
+  importer,
+  options,
+) {
+  const resolved = await this.resolve(id, importer, options)
```

Comment:
Shouldn't `skipSelf` be added, here? Interesting it is still working
```suggestion
  const resolved = await this.resolve(
    id,
    importer,
    Object.assign({ skipSelf: true }, options),
  )
```

Reply:
It's true by default in Rollup 4 so I didn't add it here.

Reply:
Ah, that explains it :)